cmake/kokkos_functions.cmake

```diff
@@ -73,7 +80,20 @@ FUNCTION(kokkos_option CAMEL_SUFFIX DEFAULT TYPE DOCSTRING)
   ELSE()
     SET(${UC_NAME} ${DEFAULT} PARENT_SCOPE)
   ENDIF()
+ENDFUNCTION()
+
+FUNCTION(kokkos_set_option CAMEL_SUFFIX VALUE)
+  LIST(FIND KOKKOS_OPTION_KEYS ${CAMEL_SUFFIX} OPTION_INDEX)
+  IF(OPTION_INDEX EQUAL -1)
+    MESSAGE(FATAL_ERROR "Couldn't set value for Kokkos_${CAMEL_SUFFIX}")
+  ENDIF()
+  SET(CAMEL_NAME Kokkos_${CAMEL_SUFFIX})
+  STRING(TOUPPER ${CAMEL_NAME} UC_NAME)
 
+  list(GET KOKKOS_OPTION_VALUES ${OPTION_INDEX} DOCSTRING)
+  list(GET KOKKOS_OPTION_TYPES ${OPTION_INDEX} TYPE)
+  SET(${CAMEL_NAME} ${VALUE} CACHE ${TYPE} ${DOCSTRING} FORCE)
```

Comment:
I think this probably makes sense for this case. I'm okay doing a cache override. I would at least print a STATUS message about what's going on.

If we don't set the cache variable, then I think what makes the most sense is error-ing out.  Print the CUDA capability detected, but force the user to specify it.